src/components/SearchOverlay.vue

```diff
@@ -246,7 +256,7 @@ async function search() {
         }
 
         predictionsList.value = (response.responses[1] as SearchTermPredictionResponse)?.predictions ?? [];
-
+        heroBanner.value = productSearchResult.value.retailMedia?.placements?.HERO_BANNER?.results![0]?.promotedDisplayAd?.result ?? undefined;
```

Comment:
Please add a short comment on top of this logic, so that a developer can read what's going on here